ci/container/build-ic.sh

```diff
@@ -82,136 +114,100 @@ export BINARIES_DIR_FULL="$ROOT_DIR/$BINARIES_DIR"
-}
-
-validate_build_env() {
-    function not_supported_prompt() {
-        echo_red "$1"
-        read -t 7 -r -s -p $'Press ENTER to continue the build anyway...\n'
-    }
-
-    if [ -n "$(git status --porcelain)" ]; then
-        echo_red "Git working directory is not clean! Clean it and retry."
-        exit 1
-    fi
-
-    if [ "$(uname)" != "Linux" ]; then
-        not_supported_prompt "This script is only supported on Linux!"
-    elif ! grep -q 'Ubuntu' /etc/os-release; then
-        not_supported_prompt "Build reproducibility is only supported on Ubuntu!"
+# Join a bash array with a string
+# https://stackoverflow.com/a/17841619
+function join_by {
+    local d=${1-} f=${2-}
+    if shift 2; then
+        printf %s "$f" "${@/#/$d}"
     fi
 }
 
-echo_green "Validating build environment"
-validate_build_env
-
 echo_blue "Purging artifact directories"
 rm -rf "$BINARIES_DIR_FULL"
 rm -rf "$CANISTERS_DIR_FULL"
 rm -rf "$DISK_DIR_FULL"
 
-echo_green "Building selected IC artifacts"
-BAZEL_CMD="bazel build --config=local --ic_version='$VERSION' --ic_version_rc_only='$IC_VERSION_RC_ONLY' \
-           --release_build=$RELEASE"
-
-BAZEL_CLEAN_CMD=$(
-    cat <<-END
-    # clear bazel cache
-    bazel clean
-END
-)
-
-BUILD_BINARIES_CMD=$(
-    cat <<-END
-    # build binaries
-    mkdir -p "$BINARIES_DIR"
-    $BAZEL_CMD //publish/binaries
-    bazel cquery --config=local --output=files //publish/binaries | xargs -I {} cp {} "$BINARIES_DIR"
-END
-)
-
-BUILD_CANISTERS_CMD=$(
-    cat <<-END
-    # build canisters
-    mkdir -p "$CANISTERS_DIR"
-    $BAZEL_CMD //publish/canisters
-    bazel cquery --config=local --output=files //publish/canisters | xargs -I {} cp {} "$CANISTERS_DIR"
-END
-)
+BAZEL_TARGETS=()
 
-BUILD_IMAGES_CMD=$(
-    cat <<-END
-    # build guestos images
-    mkdir -p "${DISK_DIR}/guestos"
-    $BAZEL_CMD //ic-os/guestos/envs/prod
-    bazel cquery --config=local --output=files //ic-os/guestos/envs/prod | xargs -I {} cp {} "${DISK_DIR}/guestos"
-    # build hostos images
-    mkdir -p "${DISK_DIR}/hostos"
-    $BAZEL_CMD //ic-os/hostos/envs/prod
-    bazel cquery --config=local --output=files //ic-os/hostos/envs/prod | xargs -I {} cp {} "${DISK_DIR}/hostos"
-    # build setupos images
-    mkdir -p "${DISK_DIR}/setupos"
-    $BAZEL_CMD //ic-os/setupos/envs/prod
-    bazel cquery --config=local --output=files //ic-os/setupos/envs/prod | xargs -I {} cp {} "${DISK_DIR}/setupos"
-END
+BAZEL_COMMON_ARGS=(
+    --config=local
+    --ic_version="$VERSION"
+    --ic_version_rc_only="$IC_VERSION_RC_ONLY"
+    --release_build="$RELEASE"
 )
-BUILD_CMD="${BAZEL_CLEAN_CMD}"
 
-if "$BUILD_BIN"; then BUILD_CMD="${BUILD_CMD}${BUILD_BINARIES_CMD}"; fi
-if "$BUILD_CAN"; then BUILD_CMD="${BUILD_CMD}${BUILD_CANISTERS_CMD}"; fi
-if "$BUILD_IMG"; then BUILD_CMD="${BUILD_CMD}${BUILD_IMAGES_CMD}"; fi
+echo_green "Building selected IC artifacts: ${BAZEL_TARGETS[*]}"
```

Comment:
Since `BAZEL_TARGETS` is initialised to `()` this will always print the empty string right? If so, what's the point of this log?

Reply:
Yep! Moved some code around and forgot to update that. Thanks!

(technically I think on bash 3 it will even crash because an empty array is considered as undefined)